Update navbar and footer navigation (#97)
tari-project/tari-dot-com: _updates/2021-12-01-update-66.md

---
layout: update
tag: Developer Update
date: 2021-12-01
author: Philip Robinson
thumbnail: update-background.jpg
title: Monero-Tari Atomic Swap RFC
class: subpage
---

Hot on the heels of the Bitcoin-Tari Atomic Swap [implementation], we have written a [RFC] proposing two methods to perform
Monero-Tari Atomic swaps. These atomic swaps are significantly more complex than a Bitcoin atomic swap because Monero

> does not include a scripting language at all. This means that the majority of the complexity of the atomic swap needs

Reviewer: ```suggestion
does not include any scripting features or appropriate time-locks. This means that the majority of the complexity of the atomic swap needs
```

to be done on the alternate chain that is being swapped with Monero.

> The RFC proposes two methods; the first is inspired by the approach taken by the [Facaster project] and [Comit network].

Reviewer 2: ```suggestion
The RFC proposes two methods; the first is inspired by the approach taken by the [Farcaster project] and [Comit network].
```

> The [RFC] goes into great detail, but the gist of the approach is when Alice wants to swap XTR for Bob's XMR. The goal
> is to produce a protocol that Alice and Bob can engage in trustlessly and that if one party stops cooperating, the other
> party can reclaim any funds that have been committed to on the respective chains.
> 

Reviewer: ```suggestion
The [RFC] goes into great detail. In short, let's assume that Alice wants to send Bob some Tari in exchange for Monero. 
```

Alice and Bob will collaborate to produce several transactions that contain aggregate keys and adaptor signatures
that become valid at a staggered timeline of lock heights. Alice and Bob will collaborate to produce all aggregate keys

> and signatures before any transaction is committed to a blockchain. The parties will then publish the transactions in an

Reviewer 3: They don't work together to do all the signatures beforehand, only the refund and lapse transactions. The swap transaction is only after publish.

Reviewer 4: removed

order that ensures both parties funds are safe at all times.

> Firstly, Alice and Bob collaborate to produce a Monero transaction locking up Bob's XMR that is spendable by an aggregate key.

Reviewer 3: This happens after the Tari UTXO is published

Reviewer 4: clarified, still want to mention this first so that its clear what is revealed by the adaptor sigs.

> Alice and Bob will collaborate to produce a Monero transaction locking up Bob's XMR that is spendable by an aggregate key.
> Alice and Bob both have a secret portion of this key, and the Monero transaction can only be spent once one party has both
> parts of the aggregate key, which will be revealed when claiming the transactions that will be placed on the Tari chain.
> This will be done after the Tari transaction is constructed as described below. The Tari transaction is constructed and 
> published first because it has multiple modes of refunding.

Reviewer: ```suggestion
The gist is that Alice and Bob collaborate to produce a Monero transaction locking up Bob's XMR that is spendable by an aggregate key.
Alice and Bob both have a secret portion of this key, and the Monero transaction can only be spent once one party has both
parts of the aggregate key. This will be revealed when claiming the transactions that will be placed on the Tari chain.

The Tari transaction is constructed and published first because it has multiple modes of refunding.
```

> Next, Alice and Bob collaborate to produce a transaction that locks up Alice's portion of the swap that can be spent by

Reviewer 3: This happens first as we can safely refund this multiple ways

> Alice and Bob collaborate to produce a transaction that locks up Alice's portion of the swap that can be spent by
> three different script keys. To spend an output for a given script key requires providing a signature that is designed
> as an adaptor signature that, if used, will reveal either Alice or Bob's portion of the aggregate key for the Monero transaction.
> The following is the TariScript that defines how the script keys can be used to spend Alice's portion of the swap.

Reviewer: ```suggestion
Alice and Bob collaborate to produce a Tari transaction that locks up Alice's portion of the swap that can be spent by
_three_ different script keys. If any of these keys are used to spend the output, information about the Monero aggregate key is revealed, allowing the correct party to claim the Monero.  Usually, it allows Alice to claim the Monero once Bob claims the Tari, but there are contingencies for Bob refunding his Monero after some expiry period.

The following is the TariScript that defines how the script keys can be used to spend Alice's portion of the swap.
```
